### PR TITLE
Add C reader thread

### PR DESCRIPTION
**Pull request by pbrook**
As per comment at top of reader_thread.c, create a separate thread for reading from the audio hardware.

### CODE REVIEW

**Review by ecobee-robbie-dumbrell — APPROVED** [review]
I believe I already tested this out on Friday per @pbrook's request (via https://github.com/ecobee/theia-go/compare/main...CAMFIRM-374/audio-overrun)

And it _appeared_ to improve the Alsa Overrun logs, so LGTM

File: alsa.go [hunk 1]
```diff
@@ -162,9 +164,16 @@ func (d *device) Close() {
...
 	}
+	if d.readerThread != nil {
+		C.reader_thread_stop(d.readerThread)
+		d.readerThread = nil
+	}
 	runtime.SetFinalizer(d, nil)
 }
 
+func (c *CaptureDevice) Close() {
+}
```
Comment (marc-evans) [inline alsa.go]: Is this needed to satisfy an interface?

File: reader_thread.c [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+        if (rc == -EPIPE) {
+            fprintf(stderr, "realtime alsa overrun");
+            s->overrun = true;
+            snd_pcm_prepare(s->h);
+        } else if (rc < 0) {
+            reader_thread_error = "snd_pcm_readi";
+            s->error = true;
+            s->stop = true;
+        } else if (s->overrun) {
+            // Drop data while waitng for _poll to clear overrun
```
Comment (marc-evans) [inline reader_thread.c]: Non-blocking typo:
```suggestion
            // Drop data while waiting for _poll to clear overrun
```

**Comment by pbrook** [comment]
For the record,should also be backwards compatible. New code is only used if `StartReadThread` is called

File: reader_thread.c [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+            s->head_offset = next_offset;
+        }
+    }
+    pthread_mutex_unlock(&s->mu);
+
+    return NULL;
+}
+
+// Returns 0 on success, 1 on overrun, -1 on error.
+// Also sets *buf_p on success;
```
Comment (marc-evans) [inline reader_thread.c]: Just `*buf`?
```suggestion
// Also sets *buf on success;
```

**Review by marc-evans — APPROVED** [review]
Logic seems sound to me. Just a few questions/comments but nothing blocking.

File: reader_thread.c (hunk 3 above)
Comment (pbrook) [reply reader_thread.c]: I changed my mind on this API half way through implementation. Comment updated.

File: alsa.go (hunk 1 above)
Comment (pbrook) [reply alsa.go]: Good spot. It's a vestigial leftover. Now removed (and retested without it!).

**Pull request merged**